x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -807,4 +817,11 @@ protected void failIndexer(String failureMessage) {
                 }, e -> {}));
         }
     }
+
+    // Considered a recoverable indexing failure
+    private static class BulkIndexFailure extends Exception {
```

Comment:
These exception classes really only need to be seen by the `isIrrecoverableFailure` method. That method currently does not know about them, but it eventually will.

Reply:
When I saw this I thought it was a bit dangerous passing it to the `onFailure` method of a listener.  If this needed to be transported to a different node then it would cause an error because the exception transport wouldn't know how to serialise it.

I think it's OK as the code stands now because it's consumed on the same node where it's thrown.

I also think it's best practice that exception class names end in `Exception`.  Certainly all the classes that extend `ElasticsearchException` follow this naming pattern.